src/jit/rangecheck.cpp

```diff
@@ -711,7 +703,7 @@ void RangeCheck::MergeEdgeAssertions(GenTreeLclVarCommon* lcl, ASSERT_VALARG_TP
                 continue;
             }
 
-            int curCns = (pRange->uLimit.IsBinOpArray()) ? pRange->uLimit.cns : 0;
+            int curCns = pRange->uLimit.cns;
```

Comment:
Unrelated but I'll note that the code below this is weird. `limit` may be `keConstant` so we can end up comparing the constant of a `keBinOpArray` with the constant of a `keConstant` to determine if `limit` tightens `pRange->uLimit`. Hmm.